### PR TITLE
fix(griffin): remove variable-length columns address/size adjustment

### PR DESCRIPTION
**Pull request by eugenels**

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: core/src/test/java/io/questdb/griffin/engine/table/DataFrameRecordCursorFactoryTest.java [hunk 1]
```diff
@@ -160,10 +168,12 @@ private void testPageFrameCursor(int rowCount, int maxSize, int startTopAt) thro
                     }
 
                     TableWriter.Row row = writer.newRow(timestamp += increment);
+                    row.putStr(strIndex, strConsts[strIndex]);
                     row.putInt(iIndex, rnd.nextInt());
                     if (startTopAt > 0 && i >= startTopAt) {
                         row.putLong(jIndex, rnd.nextLong());
-                        row.putStr(sIndex, rnd.nextChars(32));
+                        row.putStr(sIndex, strConsts[strIndex]);
+//                        row.putStr(sIndex, rnd.nextChars(32));
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/griffin/engine/table/DataFrameRecordCursorFactoryTest.java:176]: nit: this looks like a leftover.

File: core/src/test/java/io/questdb/griffin/engine/table/DataFrameRecordCursorFactoryTest.java [hunk 2]
```diff
@@ -192,14 +202,47 @@ private void testPageFrameCursor(int rowCount, int maxSize, int startTopAt) thro
...
+                        pageFrameIndex += 1;
+                        record.jumpTo(pageFrameIndex);
+                        pageAddressCache.add(pageFrameIndex, frame);
+
+                        //check variable-length columns validity
+                        for (int c = 0, n = metadata.getColumnCount(); c < n; c++) {
+                            if (ColumnType.isVariableLength(metadata.getColumnType(c))) {
+                                final long indexAddress = frame.getIndexPageAddress(c);
+                                final long size = frame.getPageSize(c);
+                                if(indexAddress > 0) {
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/griffin/engine/table/DataFrameRecordCursorFactoryTest.java:229]:
```suggestion
                                if (indexAddress > 0) {
```

File: core/src/test/java/io/questdb/griffin/engine/table/DataFrameRecordCursorFactoryTest.java [hunk 3]
```diff
@@ -192,14 +202,47 @@ private void testPageFrameCursor(int rowCount, int maxSize, int startTopAt) thro
 
                 Assert.assertTrue(factory.supportPageFrameCursor());
 
+                final CompiledFilterRecordCursor.PageFrameRecord record = new CompiledFilterRecordCursor.PageFrameRecord();
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/griffin/engine/table/DataFrameRecordCursorFactoryTest.java:205]: I'd better move this into a separate test to keep them more isolated.

File: core/src/test/java/io/questdb/griffin/engine/table/DataFrameRecordCursorFactoryTest.java [hunk 4]
```diff
@@ -192,14 +202,47 @@ private void testPageFrameCursor(int rowCount, int maxSize, int startTopAt) thro
...
+                            if (ColumnType.isVariableLength(metadata.getColumnType(c))) {
+                                final long indexAddress = frame.getIndexPageAddress(c);
+                                final long size = frame.getPageSize(c);
+                                if(indexAddress > 0) {
+                                    for (long r = 0; r < len; r++) {
+                                        final long offset = Unsafe.getUnsafe().getLong(indexAddress + r * Long.BYTES);
+                                        Assert.assertTrue(offset >= 0 && offset < size);
+                                        record.setIndex(r);
+                                        final CharSequence str = record.getStr(c);
+                                        Assert.assertEquals( strConsts[c], str);
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/griffin/engine/table/DataFrameRecordCursorFactoryTest.java:235]:
```suggestion
                                        Assert.assertEquals(strConsts[c], str);
```

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by puzpuzpuz — DISMISSED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by ideoma** [comment]
#### [PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**